### PR TITLE
Swap collateral of active loan

### PR DESCRIPTION
**Pull request by Drypto13**

### CODE REVIEW

File: contracts/modules/LoanOpenings/LoanOpenings.sol [hunk 1]
```diff
@@ -79,6 +80,55 @@ contract LoanOpenings is State, LoanOpeningsEvents, VaultController, InterestHan
...
+    {
+        require(supportedTokens[newCollateralToken], "unsupported");
+        Loan storage loanLocal = loans[loanId];
+        require(loanLocal.active, "loan is closed");
+        require(
+            msg.sender == loanLocal.borrower ||
+            delegatedManagers[loanLocal.id][msg.sender],
+            "unauthorized"
+        );
+        LoanParams storage loanParamsLocal = loanParams[loanLocal.loanParamsId];
```
Comment (RomanHiden) [inline contracts/modules/LoanOpenings/LoanOpenings.sol:100]: why not `memory` and use it as variables?

Comment (RomanHiden) [reply contracts/modules/LoanOpenings/LoanOpenings.sol:100]: or you seems to need only two variables from the struct. so you sload  only them ?

Comment (RomanHiden) [reply contracts/modules/LoanOpenings/LoanOpenings.sol:100]: actually you might be right. you have 2 sloads

Comment (Drypto13) [reply contracts/modules/LoanOpenings/LoanOpenings.sol:100]: loan params could be memory as well but i just store the values in variables anyways which are in memory.

File: contracts/modules/LoanOpenings/LoanOpenings.sol [hunk 2]
```diff
@@ -79,6 +80,55 @@ contract LoanOpenings is State, LoanOpeningsEvents, VaultController, InterestHan
...
+
+        address loanToken = loanParamsLocal.loanToken;
+        address oldCollateralToken = loanParamsLocal.collateralToken;
+        require(loanLocal.loanParamsId == keccak256(abi.encodePacked(loanToken, oldCollateralToken, true)), "not torque");
```
Comment (RomanHiden) [inline contracts/modules/LoanOpenings/LoanOpenings.sol]: you can detect torque or not by looking at maxLoanTerm instead?

Comment (RomanHiden) [inline contracts/modules/LoanOpenings/LoanOpenings.sol]: also you can use `generateLoanParamId` function

**Review by RomanHiden — APPROVED** [review]
looks good

**Pull request merged**